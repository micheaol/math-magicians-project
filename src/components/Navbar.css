nav {
  display: flex;
  justify-content: space-between;
  width: 80%;
  margin-left: 5%;
}

nav ul {
  display: flex;
  align-self: flex-start;
}

nav ul li {
  margin: 2%;
  list-style-type: none;
}
