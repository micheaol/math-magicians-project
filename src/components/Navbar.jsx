import React from 'react';
import { BrowserRouter as Router, Link } from 'react-router-dom';
import './Navbar.css';

const Navbar = () => (
  <Router>
    <nav data-testid="nav">
      <h1>Math Magician</h1>
      <ul>
        <li>
          <Link to="/">Home</Link>
        </li>
        <li>|</li>
        <li>
          <Link to="/calculator">Calculator</Link>
        </li>
        <li>|</li>
        <li>
          <Link to="/quote">Qoute</Link>
        </li>
      </ul>
    </nav>
  </Router>
);

export default Navbar;
