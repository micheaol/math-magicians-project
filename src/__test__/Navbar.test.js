import { screen, render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Navbar from '../components/Navbar';

afterEach(() => {
  cleanup();
});
describe('render the navbar to the screen', () => {
  test('render navbar', () => {
    render(<Navbar />);
    const navbarElement = screen.getByTestId('nav');
    expect(navbarElement).toBeInTheDocument();
    expect(navbarElement).toContainHTML('<h1>Math Magician</h1>');
    expect(navbarElement).toContainHTML('<li><a href="/">Home</a>');
  });

  test('matches the snapshot', () => {
    const tree = renderer.create(<Navbar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
