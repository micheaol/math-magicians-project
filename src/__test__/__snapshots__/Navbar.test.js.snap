// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the navbar to the screen matches the snapshot 1`] = `
<nav
  data-testid="nav"
>
  <h1>
    Math Magician
  </h1>
  <ul>
    <li>
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      |
    </li>
    <li>
      <a
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li>
      |
    </li>
    <li>
      <a
        href="/quote"
        onClick={[Function]}
      >
        Qoute
      </a>
    </li>
  </ul>
</nav>
`;
